sklearn/svm/base.py

```diff
@@ -779,9 +779,15 @@ def _fit_liblinear(X, y, C, fit_intercept, intercept_scaling, class_weight,
     if verbose:
         print('[LibLinear]', end='')
 
+    # LinearSVC breaks when intercept_scaling is <= 0
     bias = -1.0
     if fit_intercept:
-        bias = intercept_scaling
+        if intercept_scaling <= 0:
+            raise ValueError("Intercept scaling needs to be greater than 0."
+                             " To disable fitting an intercept,"
+                             " set fit_intercept=False.")
```

Comment:
Please put the actual value in the exception message: `"Intercept scaling is %r but needs to be greater than 0. ..." % intercept_scaling`